commit 76c75f8180a7e7592da927e8b4c8f5f067470d05
Author: Dmitry Yu Okunev <dyokunev@ut.mephi.ru>
Date:   Wed Jan 28 12:19:11 2015 +0300

    [hl_locks] A try to fix a segfault on one core PC
    
    Thanks for the bugreport to: Andrew A Savchenko <bircoph@gentoo.org>

diff --git a/privileged.c b/privileged.c
index fafd5b7..668a536 100644
--- a/privileged.c
+++ b/privileged.c
@@ -1680,11 +1680,6 @@ int privileged_init(ctx_t *ctx_p)
 #endif
 
 #ifdef CAPABILITIES_SUPPORT
-# ifdef HL_LOCKS
-	if (ncpus == 1)
-		hl_shutdown(HLLOCK_HANDLER);
-# endif
-
 	if (ctx_p->flags[SPLITTING] == SM_OFF) {
 #endif
 
@@ -1792,6 +1787,12 @@ int privileged_init(ctx_t *ctx_p)
 		default:
 			critical("Invalid ctx_p->flags[SPLITTING]: %i", ctx_p->flags[SPLITTING]);
 	}
+
+# ifdef HL_LOCKS
+	if (ncpus == 1)
+		hl_shutdown(HLLOCK_HANDLER);
+# endif
+
 	critical_on(!helper_isalive());
 
 # ifdef UNSHARE_SUPPORT
